### PR TITLE
Ad77681evb fix

### PR DESCRIPTION
**Pull request by amiclaus**
With https://github.com/analogdevicesinc/hdl/tree/fix_ad77681evb and the software changes in this PR, the AD7768-1evb project is working properly. (register configuration + capture.)

Signed-off-by: Antoniu Miclaus <antoniu.miclaus@analog.com>

### CODE REVIEW

File: projects/ad7768-1fmcz/src/ad77681evb.c [hunk 1]
```diff
@@ -81,7 +81,7 @@ struct spi_engine_init_param spi_eng_init_param  = {
 	.spi_engine_baseaddr = AD77681_SPI1_ENGINE_BASEADDR,
 	.cs_delay = 0,
 	.data_width = 32,
-	.ref_clk_hz = AD77681_SPI_ENG_REF_CLK_FREQ_HZ,
+	.ref_clk_hz = 80000000,
```
Comment (dbogdan) [inline projects/ad7768-1fmcz/src/ad77681evb.c:65]: Isn't AD77681_SPI_ENG_REF_CLK_FREQ_HZ anyway 80 MHz (https://github.com/analogdevicesinc/hdl/pull/591)?

File: projects/ad7768-1fmcz/src/ad77681evb.c [hunk 2]
```diff
@@ -100,7 +100,7 @@ struct ad77681_init_param ADC_default_init_param = {
 	AD77681_POSITIVE_FS,		// diag_mux_sel
 	false,						// conv_diag_sel
 	AD77681_CONV_16BIT,			// conv_len
-	AD77681_CRC, 				// crc_sel
+	AD77681_NO_CRC, 			// crc_sel
```
Comment (dbogdan) [inline projects/ad7768-1fmcz/src/ad77681evb.c:84]: The description of this commit doesn't contain the reason of this change. Can you provide more details?

File: projects/ad7768-1fmcz/src/ad77681evb.c [hunk 3]
```diff
@@ -135,6 +143,17 @@ int main()
 	Xil_ICacheEnable();
 	Xil_DCacheEnable();
 
+	ret = axi_clkgen_init(&clkgen, &clkgen_init);
+	if (ret != SUCCESS) {
+		printf("error: axi_clkgen_init() failed\n");
+		return FAILURE;
+	}
+
+	ret = axi_clkgen_set_rate(clkgen, 80000000);
```
Comment (mchindri) [inline projects/ad7768-1fmcz/src/ad77681evb.c:132]: If AD77681_SPI_ENG_REF_CLK_FREQ_HZ will be used, shouldn't be used here too?

File: projects/ad7768-1fmcz/src/parameters.h [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+#define __PARAMETERS_H__
+
+/******************************************************************************/
+/***************************** Include Files **********************************/
+/******************************************************************************/
+#include <xparameters.h>
+
+/******************************************************************************/
+/********************** Macros and Constants Definitions **********************/
+/******************************************************************************/
```
Comment (mchindri) [inline projects/ad7768-1fmcz/src/parameters.h:49]: Don't needed

File: projects/ad7768-1fmcz/src/ad77681evb.c (hunk 1 above)
Comment (amiclaus) [reply projects/ad7768-1fmcz/src/ad77681evb.c:65]: https://github.com/analogdevicesinc/no-OS/blob/7980ad177a0d15849116d01417318c1730bf82f7/projects/ad7768-1fmcz/src/ad77681evb.c#L64

The value inside `xparameters.h`:

image

File: projects/ad7768-1fmcz/src/ad77681evb.c (hunk 3 above)
Comment (amiclaus) [reply projects/ad7768-1fmcz/src/ad77681evb.c:132]: See above comment:

https://github.com/analogdevicesinc/no-OS/pull/795#discussion_r507604399

File: projects/ad7768-1fmcz/src/ad77681evb.c (hunk 2 above)
Comment (amiclaus) [reply projects/ad7768-1fmcz/src/ad77681evb.c:84]: Will change back to `AD77681_CRC` after #796 is merged to master.

File: projects/ad7768-1fmcz/src/parameters.h (hunk 4 above)
Comment (amiclaus) [reply projects/ad7768-1fmcz/src/parameters.h:49]: Done.

**Comment by amiclaus** [comment]
V2:
- remove duplicated header comment inside `parameters.h`

**Comment by amiclaus** [comment]
V3:
- revert back to initial driver CRC setting: `AD77681_CRC`

File: projects/ad7768-1fmcz/src/ad77681evb.c (hunk 3 above)
Comment (adrimbarean) [inline projects/ad7768-1fmcz/src/ad77681evb.c:132]: If those values are tied to each other it's probably worthwhile to have a define that we give to both.

File: projects/ad7768-1fmcz/src/parameters.h [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+#define GPIO_OFFSET							32
+
+#define GPIO_0_0							GPIO_OFFSET + 6 // 38
+#define GPIO_0_1							GPIO_OFFSET + 5 // 37
+#define GPIO_0_2							GPIO_OFFSET + 4 // 36
+#define GPIO_0_3							GPIO_OFFSET + 3 // 35
+#define GPIO_0_SYNC_IN						GPIO_OFFSET + 2 // 34
+#define GPIO_0_SYNC_OUT						GPIO_OFFSET + 1 // 33
+#define GPIO_0_RESET						GPIO_OFFSET + 0 // 32
+
```
Comment (adrimbarean) [inline projects/ad7768-1fmcz/src/parameters.h:66]: nitpick: extra line.

File: projects/ad7768-1fmcz/src/ad77681evb.c (hunk 3 above)
Comment (amiclaus) [reply projects/ad7768-1fmcz/src/ad77681evb.c:132]: I will add a define for it.

Comment (amiclaus) [reply projects/ad7768-1fmcz/src/ad77681evb.c:132]: Done.

File: projects/ad7768-1fmcz/src/parameters.h (hunk 5 above)
Comment (amiclaus) [reply projects/ad7768-1fmcz/src/parameters.h:66]: Done.

**Comment by amiclaus** [comment]
V4:
- add define for SPI Engine ref clock.
- remove extra line.

**Review by adrimbarean — APPROVED** [review]
Looks good to me.

**Pull request merged**